molecule/default/tasks/template.yml

```diff
@@ -148,20 +147,115 @@
 
+    - name: Create pods using list of template
+      kubernetes.core.k8s:
+        template:
+          - pod_template_one.j2
+          - path: "pod_template_two.j2"
+            variable_start_string: '[['
+            variable_end_string: ']]'
+          - path: "pod_template_three.j2"
+        wait: yes
+      vars:
+        k8s_pod_name_one: pod-7
+        k8s_pod_name_two: pod-8
+        k8s_pod_name_three_one: pod-9
+        k8s_pod_name_three_two: pod-10
+        k8s_pod_namespace: "template-test"
+      register: r
+
+    - name: Assert that pod creation succeeded using template
+      assert:
+        that:
+          - r is successful
+
+    # continue_on_error
+    - name: define variable for test
+      set_fact:
+        k8s_pod_name_one: pod-11
+        k8s_pod_bad_name: pod-12
+        k8s_pod_namespace: "{{ template_namespace }}"
+        k8s_pod_bad_namespace: "dummy-namespace-012345"
+
+    - name: delete pod if it exists
+      kubernetes.core.k8s:
+        template: pod_template_one.j2
+        wait: true
+        state: absent
+
+    - name: create pod on bad namespace ( continue_on_error set to default(false) )
+      kubernetes.core.k8s:
+        template:
+          - pod_with_bad_namespace.j2
+          - pod_template_one.j2
+      register: resource
+      ignore_errors: true
+
+    - name: validate that creation failed
+      assert:
+        that:
+          - resource is failed
+          - '"Failed to create object" in resource.msg'
+
+    - name: assert pod has not been created
+      kubernetes.core.k8s_info:
+        kind: "{{ item.kind }}"
+        namespace: "{{ item.namespace }}"
+        name: "{{ item.name }}"
+      with_items:
+        - kind: pod
+          namespace: "{{ k8s_pod_bad_namespace }}"
+          name: "{{ k8s_pod_bad_name }}"
+        - kind: pod
+          namespace: "{{ k8s_pod_name_one }}"
+          name: "{{ k8s_pod_namespace }}"
+      register: resource
+
+    - name: check that resources creation failed
+      assert:
+        that:
+          - '{{ resource.results[0].resources | length == 0 }}'
+          - '{{ resource.results[1].resources | length == 0 }}'
+
+    - name: create pod without namespace (continue_on_error = true)
+      kubernetes.core.k8s:
+        template:
+          - pod_with_bad_namespace.j2
+          - pod_template_one.j2
+        continue_on_error: true
+        wait: true
+      register: resource
+      ignore_errors: true
+
+    - name: validate that creation succeeded
+      assert:
+        that:
+          - resource is successful
+
+    - name: validate resource creation succeeded for some and failed for others
+      assert:
+        that:
+          - resource is successful
+          - resource_changed | length > 0
+          - resource_failed | length > 0
+      vars:
+        resource_changed: "{{ resource.result.results | json_query('[?changed==`true`]') }}"
+        resource_failed: "{{ resource.result.results | json_query('[?error]') }}"
```

Comment:
I would prefer we don't add jmespath as a test dependency just for one assertion. The same should be doable using just the builtin jinja filters. I think this should do the same:
```
assert:
  that:
    - resource.result.results | selectattr('changed') | length == 1
    - resource.result.results | selectattr('error', 'defined') | length == 1
```